Mage/src/main/java/mage/abilities/common/MayPay2LifeForColorAbility.java

```diff
@@ -0,0 +1,97 @@
+package mage.abilities.common;
+
+import mage.ObjectColor;
+import mage.abilities.Ability;
+import mage.abilities.SpellAbility;
+import mage.abilities.StaticAbility;
+import mage.abilities.costs.Cost;
+import mage.abilities.costs.common.PayLifeCost;
+import mage.abilities.costs.mana.ManaCost;
+import mage.abilities.costs.mana.ManaCosts;
+import mage.abilities.costs.mana.ManaCostsImpl;
+import mage.abilities.effects.common.cost.CostModificationEffectImpl;
+import mage.cards.Card;
+import mage.constants.CostModificationType;
+import mage.constants.Duration;
+import mage.constants.Outcome;
+import mage.constants.Zone;
+import mage.game.Game;
+import mage.players.Player;
+import mage.util.CardUtil;
+
+/**
+ * @author TheElk801
+ */
+public class MayPay2LifeForColorAbility extends StaticAbility {
+
+    private final ObjectColor color;
+
+    public MayPay2LifeForColorAbility(ObjectColor color) {
+        super(Zone.BATTLEFIELD, new MayPay2LifeEffect(color));
+        this.color = color;
+    }
+
+    private MayPay2LifeForColorAbility(final MayPay2LifeForColorAbility ability) {
+        super(ability);
+        this.color = ability.color;
+    }
+
+    @Override
+    public MayPay2LifeForColorAbility copy() {
+        return new MayPay2LifeForColorAbility(this);
+    }
+
+    @Override
+    public String getRule() {
+        return "As an additional cost to cast " + color.getDescription() +
+                " permanent spells, you may pay 2 life. Those spells cost {" + color +
+                "} less to cast if you paid life this way. This effect reduces only " +
+                "the amount of " + color.getDescription() + " mana you pay.";
+    }
+}
+
+class MayPay2LifeEffect extends CostModificationEffectImpl {
+
+    private final ObjectColor color;
```

Comment:
It seems like the only reason this is being kept is to generate the string. Can't `manaCost` be used for that purpose?

Reply:
Or better yet, generate the complete strings for the ability text and the message to ask the player in the respective constructors.

Reply:
yeah I'll just pre-make the strings

Reply:
it's needed to check the color of the spell but I'll pre-generate the ability text